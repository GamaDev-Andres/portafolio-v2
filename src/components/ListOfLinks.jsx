import propTypes from 'prop-types'

const ListOfLinks = ({ ViewInPhone = false }) => {
  return (
    <div
      className={`sm:gap-4 flex-grow sm:flex-grow-0 ${
        ViewInPhone
          ? 'sm:hidden flex w-full justify-between'
          : 'hidden justify-end sm:flex'
      }`}
    >
      <a
        className="sm:hover:bg-blue-200 flex flex-col sm:flex-row items-center transition-colors duration-300 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
        href="/#inicio"
      >
        <i className="sm:mr-2 fa-solid fa-house"></i>
        <span className={`${ViewInPhone && 'text-xs'}`}>inicio</span>
      </a>
      <a
        className="sm:hover:bg-blue-200  flex flex-col sm:flex-row items-center transition-colors duration-300 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
        href="/#sobremi"
      >
        <i className="sm:mr-2 fa-solid fa-user"></i>
        <span className={`${ViewInPhone && 'text-xs'}`}>sobre m√≠</span>
      </a>
      <a
        className="sm:hover:bg-blue-200  flex flex-col sm:flex-row items-center transition-colors duration-300 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
        href="/#habilidades"
      >
        <i className="sm:mr-2 fa-solid fa-laptop-code"></i>
        <span className={`${ViewInPhone && 'text-xs'}`}>habilidades</span>
      </a>
      <a
        className="sm:hover:bg-blue-200  flex flex-col sm:flex-row items-center transition-colors duration-300 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
        href="/#proyectos"
      >
        <i className="sm:mr-2 fa-solid fa-briefcase"></i>
        <span className={`${ViewInPhone && 'text-xs'}`}>proyectos</span>
      </a>
    </div>
  )
}
ListOfLinks.propTypes = {
  ViewInPhone: propTypes.bool
}
export default ListOfLinks
