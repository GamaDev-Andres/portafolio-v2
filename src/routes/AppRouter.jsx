import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
