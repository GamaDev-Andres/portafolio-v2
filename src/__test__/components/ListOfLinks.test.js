/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import ListOfLinks from '../../components/ListOfLinks'
import AppRouter from '../../routes/AppRouter'

describe('probando <ListOfLinks/>', () => {
  test('debe montarse correctamente', () => {
    const wrapper = Enzyme.mount(
      <ListOfLinks />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('debe montarse correctamente en telefono', () => {
    const wrapper = Enzyme.mount(
      <ListOfLinks ViewInPhone={true} />

    )

    const spans = wrapper.find('span')
    spans.forEach(span => {
      expect(span.hasClass('text-xs')).toBe(true)
    })
  })
  test('no debe de mostrarse en ruta "project"', () => {
    const wrapper = Enzyme.mount(
      <MemoryRouter initialEntries={['/project/BARBERAPP']}>
        <HelmetProvider>
          <AppRouter />
        </HelmetProvider>
      </MemoryRouter>
    )
    expect(wrapper.find('ListOfLinks').exists()).toBe(false)
  })
})
