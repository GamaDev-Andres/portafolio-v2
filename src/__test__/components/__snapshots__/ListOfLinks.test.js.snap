// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <ListOfLinks/> debe montarse correctamente 1`] = `
<div
  className="sm:gap-4 flex-grow sm:flex-grow-0 hidden justify-end sm:flex"
>
  <a
    className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
    href="/#inicio"
  >
    <i
      className="sm:mr-2 fa-solid fa-house"
    />
    <span
      className="false"
    >
      inicio
    </span>
  </a>
  <a
    className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
    href="/#sobremi"
  >
    <i
      className="sm:mr-2 fa-solid fa-user"
    />
    <span
      className="false"
    >
      sobre m√≠
    </span>
  </a>
  <a
    className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
    href="/#habilidades"
  >
    <i
      className="sm:mr-2 fa-solid fa-laptop-code"
    />
    <span
      className="false"
    >
      habilidades
    </span>
  </a>
  <a
    className="sm:hover:bg-blue-200 dark:sm:hover:bg-gris flex flex-col sm:flex-row items-center transition-colors duration-150 rounded-lg sm:px-4 sm:py-2 py-0 font-medium"
    href="/#proyectos"
  >
    <i
      className="sm:mr-2 fa-solid fa-briefcase"
    />
    <span
      className="false"
    >
      proyectos
    </span>
  </a>
</div>
`;
